waltz-test/src/main/python/waltz_ducktape/tests/validation/connection_interruption_test.py

```diff
@@ -0,0 +1,81 @@
+                 interruption_duration=10, num_interruptions=3, delay_between_interruptions=5)
+    @parametrize(num_active_partitions=4, txn_per_client=100, num_clients=2, interval=1000, timeout=300,
+                 interruption_duration=20, num_interruptions=1, delay_between_interruptions=20)
+    def test_produce_consume_with_network_interruption(self, num_active_partitions, txn_per_client, num_clients, interval, timeout,
+                                                interruption_duration, num_interruptions, delay_between_interruptions):
+        validation_cmd = self.client_cli.validate_txn_cmd(num_active_partitions, txn_per_client, num_clients, interval)
+        self.run_produce_consume_validate(lambda: self.produce_consume_with_network_interruption(validation_cmd, timeout, interruption_duration, num_interruptions, delay_between_interruptions, num_active_partitions, interval))
+
+    def drop_traffic_to_port(self, node, port):
+        node.account.ssh_capture("sudo iptables -I INPUT -p tcp --destination-port {} -j DROP".format(port))
+
+    def enable_traffic_to_port(self, node, port):
+        node.account.ssh_capture("sudo iptables -D INPUT -p tcp --destination-port {} -j DROP".format(port))
+
+    def produce_consume_with_network_interruption(self, validation_cmd, timeout, interruption_duration, num_interruptions, delay_between_interruptions, num_active_partitions, interval):
+        """
+        Set up waltz and interrupt network between a waltz client node and a server node.
+
+        :param validation_cmd: The command that is send to ClientCli
+        :param timeout: Test timeout
+        :param interruption_duration: Interval in seconds during which client won't be able to connect to server
+        :param num_interruptions: Number of connection interruption cycles
+        :param delay_between_interruptions: Interval in seconds that represents duration between network interruptions
+        :param num_active_partitions: Number of active partitions
+        :param interval: Average interval(millisecond) between transactions
+        """
+
+        partition = randrange(num_active_partitions)
+        node_idx = self.get_server_node_idx(partition)
+
+        # Start waltz cluster and wait until a storage node registers first transaction
+        self.verifiable_client.start(validation_cmd)
+        admin_port = self.waltz_storage.admin_port
+        port = self.waltz_storage.port
+        storage = self.get_host(self.waltz_storage.nodes[0].account.ssh_hostname, admin_port)
+        wait_until(lambda: self.is_max_transaction_id_updated(storage, port, partition, -1), timeout_sec=timeout)
+
+        node = self.waltz_server.nodes[node_idx]
+        for interruption in range(num_interruptions):
+            sleep(delay_between_interruptions)
+            # disable connection on port
+            self.drop_traffic_to_port(node, self.waltz_server.port)
+
+            #verify that new transactions aren't processed
```

Comment:
NIT: space after `#`
Add a command to mentioned that we choose to sleep for `interval` amount of time to make sure that all transactions that are sent to server (by the client) until then have persisted in the replicas.